test/parallel/feeLock.API.unlocking.test.ts

```diff
@@ -210,3 +211,42 @@ test("gasless- GIVEN some locked tokens and lastFeeLockBlock is lower than curre
     new BN(0)
   );
 });
+
+test("gasless- GIVEN a lock WHEN the period is N THEN the tokens can not be unlocked before that period", async () => {
+  const api = getApi();
+  let exception = false;
+
+  const feeLockAmount = JSON.parse(
+    JSON.stringify(await api.query.feeLock.feeLockMetadata())
+  ).feeLockAmount;
+  const periodLength = JSON.parse(
+    JSON.stringify(await api.query.feeLock.feeLockMetadata())
+  ).periodLength;
+
+  await testUser1.addMGATokens(sudo, new BN(feeLockAmount));
+
+  const saleAssetValue = thresholdValue.sub(new BN(5));
+
+  await testUser1.sellAssets(firstCurrency, secondCurrency, saleAssetValue);
+  const accountFeeLockData = JSON.parse(
+    JSON.stringify(
+      await api.query.feeLock.accountFeeLockData(testUser1.keyRingPair.address)
+    )
+  );
+  const waitingBlock = accountFeeLockData.lastFeeLockBlock + periodLength;
+
+  await expect(
+    unlockFee(testUser1).catch((reason) => {
+      exception = true;
+      throw new Error(reason.data);
+    })
+  ).rejects.toThrow(feeLockErrors.FeeUnlockingFail);
+  expect(exception).toBeTruthy();
```

Comment:
we need to ensure that this is still failing until endingPeriod -1, not only once :)